src/D2L.CodeStyle.Analyzers/Immutability/MutabilityInspector.cs

```diff
@@ -281,11 +287,14 @@ HashSet<ITypeSymbol> typeStack
 				}
 
 				// We descend into the base class last
-				var baseResult = InspectBaseType( type, typeStack );
-				if( baseResult.IsMutable ) {
-					return baseResult;
+				if ( type.BaseType != null ) {
+					var baseResult = InspectConcreteType( type.BaseType, typeStack );

```

Comment:
I inlined a bit of logic here: checking that `type.BaseType` isn't `null` and pulling out `type.BaseType` here rather than inside the function. If you think this would look better if there was still a `InspectBaseType` helper (that calls `InspectConcreteType`) I can make that change.